common/beerocks/bcl/source/network/network_utils.cpp

```diff
@@ -767,6 +770,54 @@ bool network_utils::linux_iface_is_up_and_running(const std::string &iface)
     return (false);
 }
 
+bool network_utils::linux_iface_get_speed(const std::string &iface, uint32_t &speed)
+{
+    int sock;
+    bool result = false;
+
+    speed = SPEED_UNKNOWN;
+
+    sock = socket(PF_INET, SOCK_DGRAM, IPPROTO_IP);
+    if (sock < 0) {
+        LOG(ERROR) << "Can't create SOCK_DGRAM socket: " << strerror(errno);
+    } else {
+        struct ifreq ifr;
+        struct ethtool_link_settings settings;
+        int rc;
+
+        beerocks::string_utils::copy_string(ifr.ifr_name, iface.c_str(), sizeof(ifr.ifr_name));
+        ifr.ifr_data = reinterpret_cast<char *>(&settings);
+
+        settings.cmd = ETHTOOL_GSET;
```

Comment:
Unless I'm very mistaken, this must be ETHTOOL_GLINKSETTINGS. Now you're doing two times the same thing but with different structs...

Reply:
You were right. However changing it to ETHTOOL_GLINKSETTINGS was not enough (such change has uncovered a bug that I've had to fix too).